### PR TITLE
client: add support for passing 'ack_timeout' and add is_alive to client api

### PR DESCRIPTION
**Pull request by assafgi**
…

### CODE REVIEW

File: tests/uts/test_client.py [hunk 1]
```diff
@@ -189,3 +190,12 @@ def test_pid_not_found(self):
...
         self.assertRaises(CommandAlreadyDone, cmd.get_pid)
+
+    def test_ack_timeout(self):
+        cmd = self.client.run(self.host_id, 'bash', '-ce', 'sleep 20', ack_timeout=1)
+        with self.assertRaises(TalkerServerTimeout) as exc:
+            cmd.result()
+
+        # We expect to have about 6 seconds timeout
+        # When we don't get ack we call is_alive that has a 5 seconds timeout
+        assert int(exc.exception.timeout) == 6
```
Comment (DoronCohen) [inline tests/uts/test_client.py]: This looks fragile

File: talker/command.py [hunk 2]
```diff
@@ -199,7 +201,7 @@ def send(self):
 
     def on_sent(self):
         if not self.ack_timer:
-            self.ack_timer = Timer(expiration=TALKER_CONTEXT.ack_timeout)  # this is expiration on the agent ack'ing the command
```
Comment (DoronCohen) [inline talker/command.py]: Is this context used anywhere in weka's code?

File: talker/command.py [hunk 3]
```diff
@@ -438,7 +440,7 @@ def check_client_timeout(self):
                 talker_alive = False
                 if self.name != is_talker_alive_name:  # prevent recursion
                     try:
-                        self.talker.run(self.host_id, 'true', name=is_talker_alive_name, timeout=5).wait()
+                        self.talker.is_alive(self.host_id, name=is_talker_alive_name, timeout=5).wait()
```
Comment (DoronCohen) [inline talker/command.py]: `is_alive` will not raise an exception. We should add a test for this scenario

File: talker/client.py [hunk 4]
```diff
@@ -216,6 +218,16 @@ def is_done(self, cmds):
...
         return len(list(filter(lambda i: i is None, self.poll(cmds)))) == 0
 
+    def is_alive(self, host_id, name='is_alive', timeout=5):
+        """
+        Do a naive check to see if the agent is responsive.
+        :return: True if we get ack, False otherwise
+        If the function returns False it doesn't mean for sure that the agent/machine is down
+        """
+        cmd = self.run(host_id, 'true', name=name, timeout=timeout, ack_timeout=5)
+        cmd.wait()
```
Comment (DoronCohen) [inline talker/client.py]: When will this wait return?

File: talker/client.py [hunk 5]
```diff
@@ -216,6 +218,16 @@ def is_done(self, cmds):
         """
         return len(list(filter(lambda i: i is None, self.poll(cmds)))) == 0
 
+    def is_alive(self, host_id, name='is_alive', timeout=5):
+        """
+        Do a naive check to see if the agent is responsive.
+        :return: True if we get ack, False otherwise
+        If the function returns False it doesn't mean for sure that the agent/machine is down
```
Comment (DoronCohen) [inline talker/client.py]: Please document the parameters the function receives

File: talker/command.py (hunk 2 above)
Comment (assafgi) [reply talker/command.py]: Yes...

File: talker/client.py (hunk 4 above)
Comment (assafgi) [reply talker/client.py]: when ack is timed out or command is timed out.

File: talker/client.py [hunk 6]
```diff
@@ -216,6 +218,23 @@ def is_done(self, cmds):
...
         return len(list(filter(lambda i: i is None, self.poll(cmds)))) == 0
 
+    def is_alive(self, host_id, name='is_alive'):
+        """
+        Do a naive check to see if the agent is responsive.
+
+        :param [str] host_id: The ID of the host to send the command to
+        :param [str] name: A name to give to the command (for logging pruposes)
+        :return: True if we get ack, False otherwise
+         If the function returns False it doesn't mean for sure that the agent/machine is down
```
Comment (DoronCohen) [inline talker/client.py]: This line should be above the parameters doc

**Pull request merged**